presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/TransformExistsApplyToLateralNode.java

```diff
@@ -38,19 +43,33 @@
 import java.util.Optional;
 
 import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
 import static com.facebook.presto.sql.planner.plan.LateralJoinNode.Type.INNER;
+import static com.facebook.presto.sql.planner.plan.LateralJoinNode.Type.LEFT;
 import static com.facebook.presto.sql.planner.plan.Patterns.applyNode;
+import static com.facebook.presto.sql.tree.BooleanLiteral.TRUE_LITERAL;
 import static com.facebook.presto.sql.tree.ComparisonExpressionType.GREATER_THAN;
+import static com.google.common.base.Preconditions.checkState;
 import static com.google.common.collect.Iterables.getOnlyElement;
 import static java.util.Objects.requireNonNull;
 
 /**
  * Exists is modeled as:
  * <pre>
- *     - Project($0 > 0)
- *       - Aggregation(COUNT(*))
- *         - Limit(1)
- *           -- subquery
+ *     - LateralJoin(LEFT)
+ *       - input
+ *       - Project($0 > 0)
+ *         - Aggregation(COUNT(*))
+ *           - subquery
+ * </pre>
+ * or (if correlated predicates are equality comparisons):
+ * <pre>
+ *     - Project(exits := COALESCE(subqueryTrue, false))
```

Comment:
this should go above, because you first try to use this rewrite and then you fallback to above default rewrite.